integr8ly/installation
# INTLY-1260 Add a var for *not* running master tasks

This change allows for running an install without the need to
ssh to the master(s).
In this scenario, certain extra vars should be set, and other tasks
done outside of the install run.


## Verification Steps

To verify on PDS (can ssh to master during install), run an install as normal. Everything should behave as it did before. As should uninstall.

To verify on OSD (*no* ssh to master during install) do the following:

- run the install.yml playbook:
  - based on the managed.template inventory file
  - on a machine that is logged in (oc) with cluster-admin access (*not* system:admin user)
  - with values set for the following vars
    - eval_app_host (router subdomain, used for creating some routes & for 3scale)
    - openshift_master_url (master url for oauth use)
- after install, add the relevant cors rule to the master-config.yaml based on the webapp route
  - Get the webapp host using `oc get route tutorial-web-app -n webapp --template 'https://{{.spec.host}} '`
  - Add the following to master-config.yaml
```
   corsAllowedOrigins:
   - <webapp_host>
```
- add the cluster SSO identity provider to the master-config.yaml
  - Get the client secret using `oc get secret openshift-client-client -n sso --template '{{.data.install}}' | base64 --decode`
  - Get the sso host using `oc get route sso -n sso --template 'https://{{.spec.host}} '`
  - Add the following to master-config.yaml
```
   identityProviders:
   - name: rh_sso
     challenge: true
     login: true
     mappingInfo: add
     provider:
       apiVersion: v1
       kind: OpenIDIdentityProvider
       clientID: openshift-client
       clientSecret: <client_secret>
       ca: ca.crt
       urls:
         authorize: <sso_host>/auth/realms/openshift/protocol/openid-connect/auth
         token: <sso_host>/auth/realms/openshift/protocol/openid-connect/token
         userInfo: <sso_host>/auth/realms/openshift/protocol/openid-connect/userinfo
         claims:
         id:
           - sub
         preferredUsername:
           - preferred_username
          name:
           - name
         email:
           - email
```
- restart the master(s)
- you should be able to login as normal to openshift & the various services. The webapp should behave as normal (i.e. provisions things for your user by calling the openshift api from browser)
